### PR TITLE
Add sitemap.xml and robots.txt

### PR DESCRIPTION
**Pull request by mheap**
Signed-off-by: Michael Heap <m@michaelheap.com>

Add missing `sitemap.xml` and `robots.txt`

Did you sign your commit? Yes

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)? Yes

### CODE REVIEW

File: app/_plugins/generators/sitemap.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+module Sitemap
+  class Generator < Jekyll::Generator # rubocop:disable Metrics/ClassLength
+    priority :lowest
+
+    def generate(site)
+      # What's our latest version?
+      latest = site.data['versions'].reject { |v| v['release'] == 'dev' }.last
```
Comment (lahabana) [inline app/_plugins/generators/sitemap.rb:7]: versions.yml has lost the `latest` boolean that used to exist. This `latest` boolean is super useful as it enables us to ship the docs for the release during stabilization week without making it indexed by Google and still show the real current one.

In other words as soon as we'd cut the 2.0 release the 2.0 docs would start to exist but `latest` on the website would still be `1.8`.
Does that make sense?

File: app/robots.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+User-agent: * 
+Disallow: /latest_version 
+Disallow: /latest_version.html 
+  
+Sitemap: https://kuma.io/sitemap.xml
```
Comment (lahabana) [inline app/robots.txt:5]: IIRC the robots.txt was pretty simple but there was a bunch a meta on the pages of non latest docs to skip indexing them.

**Review by lahabana — COMMENTED** [review]
I think latest was useful feature

File: app/_plugins/generators/sitemap.rb (hunk 1 above)
Comment (lahabana) [reply app/_plugins/generators/sitemap.rb:7]: This was discussed live and we decided to do this in the next PR :)

File: app/_plugins/generators/sitemap.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        "/install/",
+        "/docs/",
+      ]
+      versioned.each do |v|
+        return true if url.include?(v)
+      end
+      false
+    end
+
+    def is_latest(url, latest)
```
Comment (lahabana) [inline app/_plugins/generators/sitemap.rb:63]: nit: this is more like `is_of_version(url, version)` as the latest param could be any version in practice.

**Review by lahabana — APPROVED** [review]
LGTM! There's a question but no reason to block this PR!

File: app/_plugins/generators/sitemap.rb (hunk 1 above)
Comment (mheap) [reply app/_plugins/generators/sitemap.rb:7]: It makes sense - will re-add `latest` in a followup PR

File: app/robots.txt (hunk 2 above)
Comment (mheap) [reply app/robots.txt:5]: Will add this in a followup

Comment (mheap) [reply app/robots.txt:5]: I've just investigated and the metadata is present on non-latest docs. They're set to `noindex,follow` which is correct

**Pull request merged**